libraries/Microsoft.Bot.Builder/Exceptions/BotStateException.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+
+namespace Microsoft.Bot.Builder.Exceptions
```

Comment:
I don't think we need to create a custom exception for this.

Reply:
Are you saying we should use the base `Exception` type or are you saying there's an existing exception type that would work here?